src/Corax/Queries/TermProviders/TermProvider.StartWith.cs

```diff
@@ -14,36 +14,50 @@ namespace Corax.Queries
     [DebuggerDisplay("{DebugView,nq}")]
     public struct StartWithTermProvider : ITermProvider
     {
+        private readonly CompactTree _tree;
         private readonly IndexSearcher _searcher;
-        private CompactTree.Iterator _iterator;
         private readonly FieldMetadata _field;
         private readonly Slice _startWith;
-        private readonly CompactTree _tree;
+
+        private CompactTree.Iterator _iterator;
+
         public StartWithTermProvider(IndexSearcher searcher, CompactTree tree, FieldMetadata field, Slice startWith)
         {
             _searcher = searcher;
             _field = field;
             _iterator = tree.Iterate();
             _startWith = startWith;
             _tree = tree;
-            
+
             _iterator.Seek(_startWith);
         }
 
         public void Reset() => _iterator.Seek(_startWith);
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public bool Next(out TermMatch term) => Next(out term, out _);
+        public bool Next(out TermMatch term)
+        {
+            var result = Next(out term, out var scope);
+            scope.Dispose();
+            return result;
+        } 
 
-        public bool Next(out TermMatch term, out Slice termSlice)
+        public bool Next(out TermMatch term, out CompactKeyCacheScope termScope)
         {
-            if (_iterator.MoveNext(out termSlice, out var _) == false || termSlice.StartWith(_startWith) == false)
+            if (_iterator.MoveNext(out termScope, out var _) == false)
+            {
+                term = TermMatch.CreateEmpty(_searcher, _searcher.Allocator);
+                return false;
+            }
+
+            var key = termScope.Key.Decoded();
+            if (key.StartsWith(_startWith) == false)
```

Comment:
Can't we do this on the encoded value?

Reply:
Most likely we can, I still have to figure out how.